packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/ProviderVirtualMachines.tsx

```diff
@@ -51,12 +60,16 @@ interface ProviderVirtualMachinesProps extends RouteComponentProps {
   name?: string;
   loaded?: boolean;
   loadError?: unknown;
+  rowMapper?: React.FunctionComponent<RowProps<VmData>>;
+  fieldsMetadataFactory?: ResourceFieldFactory;
 }
 
 export const ProviderVirtualMachines: React.FC<ProviderVirtualMachinesProps> = ({
   obj,
   loaded,
   loadError,
+  rowMapper = ProviderVirtualMachinesRow,
+  fieldsMetadataFactory = vmsFieldsMetadataFactory,
 }) => {
   const { t } = useForkliftTranslation();
   const [userSettings] = useState(() => loadUserSettings({ pageId: 'ProviderVMs' }));
```

Comment:
please move all the hooks to the individual lists, e.g. the openshiftvmlist, ovirtvmslit ....
we will have time to do refactoring later

Reply:
hooks should move to providers lists, each provider should fetch it's own data